src/code/views/dropdown-view.coffee

```diff
@@ -54,27 +56,22 @@ DropDown = React.createClass
 
   componentWillMount: ->
     if window.addEventListener
-      window.addEventListener 'mousedown', @checkBlur, true
-      window.addEventListener 'touchstart', @checkBlur, true
+      window.addEventListener 'mousedown', @checkClose, true
+      window.addEventListener 'touchstart', @checkClose, true
 
   componentWillUnmount: ->
     if window.removeEventListener
-      window.removeEventListener 'mousedown', @checkBlur, true
-      window.removeEventListener 'touchstart', @checkBlur, true
-
-  checkBlur: ->
-    if @state.showingMenu
-      @blur()
-
-  blur: ->
-    @unblur()
-    timeout = setTimeout ( => @setState {showingMenu: false, subMenu: false} ), 500
-    @setState {timeout: timeout}
-
-  unblur: ->
-    if @state.timeout
-      clearTimeout(@state.timeout)
-    @setState {timeout: null}
+      window.removeEventListener 'mousedown', @checkClose, true
+      window.removeEventListener 'touchstart', @checkClose, true
+
+  checkClose: (evt) ->
+    # if the click is on the menu, let the menu handle it
```

Comment:
Since this is called for every click in the window I wonder if we should short circuit and exit if `@state.showingMenu` is `false` - that will avoid the walking the DOM tree for each click.

Reply:
Excellent point! Done.